packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx

```diff
@@ -166,20 +201,27 @@ const SchedulerFilters: FC<SchedulerFiltersProps> = ({
         >
             {dashboard && dashboard.filters.dimensions.length > 0 ? (
                 <Stack>
-                    {dashboard?.filters?.dimensions.map((filter) => (
-                        <FilterItem
-                            key={filter.id}
-                            dashboardFilter={filter}
-                            schedulerFilter={schedulerFilters.find(
-                                (f) =>
-                                    f.target.fieldId ===
-                                        filter.target.fieldId &&
-                                    f.target.tableName ===
-                                        filter.target.tableName,
-                            )}
-                            onChange={handleUpdateSchedulerFilter}
-                        />
-                    ))}
+                    {dashboard?.filters?.dimensions.map((filter) => {
+                        const schedulerFilter =
+                            schedulerFiltersData && schedulerFiltersData.length
+                                ? schedulerFiltersData.find(
+                                      (f) =>
+                                          f.target.fieldId ===
+                                              filter.target.fieldId &&
+                                          f.target.tableName ===
+                                              filter.target.tableName,
+                                  )
```

Comment:
In `applyDimensionOverrides` and in line 130 in this file, we use the filter id to find a match. Here we use the field ID  and table name. 

This will be a problem if:

- create dashboard with filter A ( with ID X + field ID 1 and table name W)
- create scheduler with filter B (with ID X + field ID 1 and table name W )  --> scheduler filter is valid
- update dashboard with no filters  --> scheduler filter is invalid
- update dashboard with filter C (with ID Y + field ID 1 and table name W) --> scheduler filter is shown since field id and table name match, but onChange returns undefined since the filter ids don't match

Reply:
Ah, I see. Makes total sense to not use this `fieldId` and `tableName` - and to avoid dashboard filters updates + changes to fields with diff operators, values, that could cause unexpected results. Changing it to search by `id` so we ensure uniqueness!